sky/clouds/nebius.py

```diff
@@ -247,6 +247,18 @@ def make_deploy_resources_variables(
 
         use_static_ip_address = skypilot_config.get_nested(
             ('nebius', 'use_static_ip_address'), default_value=False)
+
+        # Add Nebius-specific setup commands that run during provisioning
+        initial_setup_commands = []
+
+        # This adds an /etc/hosts entry mapping the node's IP to its
+        # hostname, which is useful for distributed training frameworks that
+        # use MASTER_ADDR=$(hostname) patterns.
+        hostname_setup_cmd = (
+            'IP=$(hostname -I | awk \'{print $1}\'); '
+            'echo "$IP $(hostname)" | sudo tee -a /etc/hosts; ')
+        initial_setup_commands.append(hostname_setup_cmd)
+
```

Comment:
Any reason we add it through inital_setup_commands, instead of directly adding to the template for setup?